--- freeswitch/libs/libsndfile/src/Makefile.am.orig	2010-05-24 22:37:21.000000000 +0200
+++ freeswitch/libs/libsndfile/src/Makefile.am	2010-05-24 22:48:34.000000000 +0200
@@ -20,7 +20,7 @@
 
 noinst_HEADERS = common.h sfconfig.h sfendian.h wav_w64.h sf_unistd.h
 
-noinst_PROGRAMS = test_main
+noinst_PROGRAMS =
 
 COMMON = common.c file_io.c command.c pcm.c ulaw.c alaw.c float32.c \
 		double64.c ima_adpcm.c ms_adpcm.c gsm610.c dwvw.c vox_adpcm.c \
@@ -40,15 +40,15 @@
 
 libcommon_la_SOURCES = $(COMMON)
 
-test_main_SOURCES = test_main.c test_main.h test_conversions.c test_float.c test_endswap.c \
-					test_audio_detect.c test_log_printf.c test_file_io.c test_ima_oki_adpcm.c
-test_main_LDADD = libcommon.la
+#test_main_SOURCES = test_main.c test_main.h test_conversions.c test_float.c test_endswap.c \
+#					test_audio_detect.c test_log_printf.c test_file_io.c test_ima_oki_adpcm.c
+#test_main_LDADD = libcommon.la
+#
+#
+#test_endswap.c: test_endswap.def test_endswap.tpl
+#	autogen --writable test_endswap.def
 
-
-test_endswap.c: test_endswap.def test_endswap.tpl
-	autogen --writable test_endswap.def
-
-genfiles : test_endswap.c $(SYMBOL_FILES)
+genfiles : $(SYMBOL_FILES)
 
 # A single test programs.
 # It is not possible to place these in the tests/ directory because they
