diff -ur libdingaling/Makefile.am libdingaling.new/Makefile.am
--- libdingaling/Makefile.am	2008-04-10 03:22:25.000000000 +0000
+++ libdingaling.new/Makefile.am	2008-04-14 01:02:01.288891817 +0000
@@ -8,33 +8,18 @@
 AM_CFLAGS   = $(new_AM_CFLAGS) -I./src -I$(prefix)/include
 AM_CPPFLAGS = $(AM_CFLAGS)
 AM_LDFLAGS  = $(new_AM_LDFLAGS)
-
-AM_CFLAGS  += `../apr/apr-1-config --cflags --cppflags --includes || $(prefix)/bin/apr-1-config --cflags --cppflags --includes`
-AM_CFLAGS  += `../apr-util/apu-1-config --includes || $(prefix)/bin/apu-1-config --includes`
-AM_CFLAGS  += -I../iksemel/include
+AM_LIBADD   = $(new_AM_LIBADD)
 
 lib_LTLIBRARIES		= libdingaling.la
 libdingaling_la_SOURCES	= src/libdingaling.c src/sha1.c
 libdingaling_la_CFLAGS	= $(AM_CFLAGS)
 libdingaling_la_LDFLAGS	= $(AM_LDFLAGS)
-libdingaling_la_LIBADD  = ../iksemel/src/libiksemel.la ../apr/libapr-1.la ../apr-util/libaprutil-1.la
+libdingaling_la_LIBADD  = $(AM_LIBADD)
 library_includedir	= $(prefix)/include
 library_include_HEADERS = src/libdingaling.h
 
 $(libdingaling_la_SOURCES): $(libdingaling_la_LIBADD)
 
-../apr/libapr-1.la: ../apr ../apr/.update
-	cd ../apr && $(MAKE)
-	$(TOUCH_TARGET)
-
-../apr-util/libaprutil-1.la: ../apr-util ../apr-util/.update
-	cd ../apr-util && $(MAKE)
-	$(TOUCH_TARGET)
-
-../iksemel/src/libiksemel.la: ../iksemel ../iksemel/.update
-	cd ../iksemel && $(MAKE)
-	$(TOUCH_TARGET)
-
 dox:
 	cd docs && doxygen $(PWD)/docs/Doxygen.conf
 
diff -ur libdingaling/configure.in libdingaling.new/configure.in
--- libdingaling/configure.in	2008-04-10 03:22:25.000000000 +0000
+++ libdingaling.new/configure.in	2008-04-14 01:09:39.918889344 +0000
@@ -118,8 +118,17 @@
     fi
 fi
 
+PKG_CHECK_MODULES([iksemel],[iksemel])
+new_AM_CFLAGS="$new_AM_CFLAGS $iksemel_CFLAGS"
+new_AM_LIBADD="$new_AM_LIBADD $iksemel_LIBS"
+
+PKG_CHECK_MODULES([apr],[apr-1])
+new_AM_CFLAGS="$new_AM_CFLAGS $apr_CFLAGS"
+new_AM_LIBADD="$new_AM_LIBADD $apr_LIBS"
+
 AC_SUBST(new_AM_CFLAGS)
 AC_SUBST(new_AM_LDFLAGS)
+AC_SUBST(new_AM_LIBADD)
 AC_SUBST(SOLINK)
 AC_SUBST(DYNAMIC_LIB_EXTEN)
 AC_DEFINE_UNQUOTED([__inline__],[$IN_LINE],[sunpro is bad at inline])
