--- configure.ac	2016-05-05 22:42:20.000000000 +0200
+++ configure.ac	2016-06-08 16:28:26.158687444 +0200
@@ -1365,7 +1365,9 @@
   AM_CONDITIONAL([HAVE_ILBC],[true])],[
     PKG_CHECK_MODULES([ILBC], [ilbc >= 0.0.1],[
       AM_CONDITIONAL([HAVE_ILBC],[true])],[
-      AC_MSG_RESULT([no]); AM_CONDITIONAL([HAVE_ILBC],[false])])])
+        PKG_CHECK_MODULES([ILBC], [libilbc >= 0.0.1],[
+          AM_CONDITIONAL([HAVE_ILBC],[true])],[
+          AC_MSG_RESULT([no]); AM_CONDITIONAL([HAVE_ILBC],[false])])])])
 
 PKG_CHECK_MODULES([G7221], [g722_1 >= 0.2.0],[
   AM_CONDITIONAL([HAVE_G7221],[true])],[
