--- wanpipe-3.5.20/api/libstelephony/Makefile.am	2010-09-07 21:10:25.000000000 +0200
+++ wanpipe-3.5.20.17/api/libstelephony/Makefile.am	2011-05-24 23:10:51.000000000 +0200
@@ -1,6 +1,6 @@
  
 WLINC=../../patches/kdrivers/include
-WINC=$(DESTDIR)/usr/include/wanpipe
+WINC=/usr/include/wanpipe
 
 AM_CFLAGS		= -fPIC -Wall -Wstrict-prototypes -Wmissing-prototypes -g -I$(WLINC) -I$(WINC)  -I. \
 	-D__LINUX__ -D_REENTRANT -D_GNU_SOURCE -O2 -D_DEBUG_=2 -D_GNUC_ -I../lib -Istel_tone 
@@ -8,16 +8,13 @@ AM_CFLAGS		= -fPIC -Wall -Wstrict-protot
 AM_CXXFLAGS		= -fPIC -Wall -g -I$(WLINC) -I$(WINC)  -I. \
 	-D__LINUX__ -D_REENTRANT -D_GNU_SOURCE -O2 -D_DEBUG_=2 -D_GNUC_ -I../lib  -Istel_tone
 
-LIB_SOUCES		= libstelephony.cpp PToneDecoder.cpp PToneEncoder.cpp Q931EventsDecoder.cpp stelephony.cpp stel_tone/stel_tone.c stel_tone/wp_uart.c stel_tone/wp_libteletone_generate.c stel_tone/wp_libteletone_detect.c stel_tone/wp_fsk.c
 
+lib_LTLIBRARIES = libstelephony.la
+include_HEADERS = libstelephony.h 
 
-library_includedir      = $(includedir)
-library_include_HEADERS = libstelephony.h 
+libstelephony_la_SOURCES = libstelephony.cpp PToneDecoder.cpp PToneEncoder.cpp Q931EventsDecoder.cpp stelephony.cpp stel_tone/stel_tone.c stel_tone/wp_uart.c stel_tone/wp_libteletone_generate.c stel_tone/wp_libteletone_detect.c stel_tone/wp_fsk.c
 
-lib_LTLIBRARIES         = libstelephony.la
-libstelephony_la_CXXFLAGS    = $(AM_CXXFLAGS)
-libstelephony_la_LDFLAGS   = -version-info @LIBSTELEPHONY_LT_CURRENT@:@LIBSTELEPHONY_LT_REVISION@:@LIBSTELEPHONY_LT_AGE@
-libstelephony_la_SOURCES   = $(LIB_SOUCES)
+libstelephony_la_CFLAGS    = $(AM_CFLAGS)
+libstelephony_la_CXXFLAGS  = $(AM_CXXFLAGS)
+libstelephony_la_LDFLAGS   = -version-info 1:0:0
 
-lib_LIBRARIES           = libstelephony.a
-libstelephony_a_CXXFLAGS     = $(AM_CXXFLAGS)
--- wanpipe-3.5.20/api/libstelephony/configure.in	2009-08-25 22:44:41.000000000 +0200
+++ wanpipe-3.5.20.17/api/libstelephony/configure.in	2011-05-24 23:06:13.000000000 +0200
@@ -2,36 +2,18 @@
 # Process this file with autoconf to produce a configure script.
 
 AC_PREREQ(2.59)
-AC_INIT(libstelephony, 2.0.0, anthmct@yahoo.com)
-AM_INIT_AUTOMAKE(libstelephony,2.0.0)
-
-LIBSTELEPHONY_LT_CURRENT=2  # interface 2
-LIBSTELEPHONY_LT_REVISION=0 # first revision of this interface
-LIBSTELEPHONY_LT_AGE=0      #not backwards compatible (0 previous interfaces are compatible)
-
-#AC_CONFIG_MACRO_DIR([m4])
-
-AC_SUBST(LIBSTELEPHONY_LT_CURRENT)
-AC_SUBST(LIBSTELEPHONY_LT_REVISION)
-AC_SUBST(LIBSTELEPHONY_LT_AGE)
+AC_INIT(libstelephony, 2.0.0)
+AC_CONFIG_HEADERS(config.h)
+AM_INIT_AUTOMAKE
 
 # Checks for programs.
 AC_PROG_CC
 AC_PROG_CXX
 AC_PROG_LIBTOOL
-AC_CONFIG_HEADERS(config.h)
 
-AC_CHECK_PROG(AR, ar, ar, no)
 # Checks for libraries.
-
 AC_CHECK_HEADERS(fcntl.h netinet/in.h stdlib.h string.h sys/ioctl.h sys/socket.h sys/time.h unistd.h math.h)
 
-AC_ARG_WITH(libpri,
-	[  --with-libpri=<path> enable pri support],[libpripath="$withval"])
-
-AM_CONDITIONAL([LIBPRI],[test -d "$libpripath" ])
-AC_SUBST([libpripath])
-
 # Checks for typedefs, structures, and compiler characteristics.
 AC_HEADER_TIME
 
